pallets/market/src/lib.rs

```diff
@@ -268,9 +272,9 @@ pub mod pallet {
             )
         }
 
-        fn cid(&self) -> Result<Cid, ProposalError> {
+        fn cid(&self) -> Result<Cid, PaddedPieceSizeError> {
             let cid = Cid::try_from(&self.piece_cid[..])
-                .map_err(|e| ProposalError::InvalidPieceCid(e))?;
+                .map_err(|_| PaddedPieceSizeError::InvalidPieceCid)?;
```

Comment:
Why can't this one be 
```suggestion
                .map_err(|_| Error::<T>::InvalidPieceCid)?;
```